crates/daphne/src/pine/flp.rs

```diff
@@ -360,10 +354,8 @@ impl<F: FftFriendlyFieldElement> Type for PineType<F> {
     type Field = F;
 
     fn encode_measurement(&self, _measurement: &()) -> Result<Vec<F>, FlpError> {
-        // Pine can't use this feature of `Type` because the encoding takes an input called the
-        // "wraparound joint randomness" that is derived from secret shares of an intermediate
-        // representation of the input.
-        unreachable!()
+        // PINE uses a randomized encoding, where the randomness is derived by the protocol.
+        unimplemented!("use encode_with_wr_joint_rand() instead")
     }
```

Comment:
if it can't implement this method than it shouldn't implement this trait :thinking: is there a way to rethink this?

Reply:
This unfortunately is required by the API upstream in the `prio` crate: https://github.com/divviup/libprio-rs/issues/1108